### PR TITLE
docs: add deprecation section

### PR DESCRIPTION
**Pull request by 317brian**
Add deprecation section at the top of the release notes.

Once the migration guide for MVD -> array is merged, we can add a link to it.

@ektravel do you think we should add it to the upgrade notes too?

### CODE REVIEW

File: docs/release-info/release-notes.md [hunk 1]
```diff
@@ -53,6 +53,18 @@ If your release note contains images, put the images in the release-info/assets
...
 -->
 
+## Upcoming deprecations
+
+As part of the continued improvements to Druid, we are deprecating certain features and behaviors in favor of newer iterations that offer more robust features and are more aligned with standard ANSI SQL. Many of these new features have been the default for new deployments for several releases.
+
+The following features are deprecated, and we plan to remove support as early as Druid 36.0.0:
+
+* **Multi-value dimensions**: Druid now offers support for [array types](../querying/sql-data-types.md#arrays). 
+* **Non-SQL compliant null handling**: By default, Druid now differentiates between an empty string and a record with no data as well as between an empty numerical record and `0`. For more information, see [NULL values](../querying/sql-data-types.md#null-values). For a tutorial on the  SQL-compliant logic, see the [Null handling tutorial](../tutorials/tutorial-sql-null.md).
```
Comment (ektravel) [inline docs/release-info/release-notes.md:63]:
```suggestion
* **Non-SQL compliant null handling**: By default, Druid now differentiates between an empty string and a record with no data as well as between an empty numerical record and `0`. For more information, see [NULL values](../querying/sql-data-types.md#null-values). For a tutorial on the SQL-compliant logic, see the [Null handling tutorial](../tutorials/tutorial-sql-null.md).
```

File: docs/release-info/release-notes.md [hunk 2]
```diff
@@ -53,6 +53,18 @@ If your release note contains images, put the images in the release-info/assets
...
+* **Non-strict Boolean handling**: Druid now strictly uses `1` (true) or `0` (false). Previously, true and false could be represented either as `true` and `false` as well as `1` and `0`, respectively. In addition, Druid now returns a null value for Boolean comparisons like `True && NULL`. [Boolean logic](../querying/sql-data-types.md#boolean-logic). For examples of filters that use the SQL-compliant logic, see [Query filters](../querying/filters.md).
```
Comment (ektravel) [inline docs/release-info/release-notes.md:64]:
```suggestion
* **Non-strict Boolean handling**: Druid now strictly uses `1` (true) or `0` (false). Previously, true and false could be represented either as `true` and `false` as well as `1` and `0`, respectively. In addition, Druid now returns a null value for Boolean comparisons like `True && NULL`. For more information, see [Boolean logic](../querying/sql-data-types.md#boolean-logic). For examples of filters that use the SQL-compliant logic, see [Query filters](../querying/filters.md).
```

File: docs/release-info/release-notes.md [hunk 3]
```diff
@@ -53,6 +53,18 @@ If your release note contains images, put the images in the release-info/assets
...
+* **Two-value logic**: By default, Druid now uses three-valued logic for both ingestion and querying. This primarily affects filters using logical NOT operations on columns with NULL values.  For more information, see [Boolean logic](../querying/sql-data-types.md#boolean-logic). For examples of filters that use the SQL-compliant logic, see [Query filters](../querying/filters.md).
```
Comment (ektravel) [inline docs/release-info/release-notes.md:65]:
```suggestion
* **Two-value logic**: By default, Druid now uses three-valued logic for both ingestion and querying. This primarily affects filters using logical NOT operations on columns with NULL values. For more information, see [Boolean logic](../querying/sql-data-types.md#boolean-logic). For examples of filters that use the SQL-compliant logic, see [Query filters](../querying/filters.md).
```

**Review by ektravel — COMMENTED** [review]
Having two consecutive headings with the word "deprecations" can be confusing. Also calling them "upcoming" is misleading since the features are already deprecated. We should consider different headings to clearly differentiate the two sections. For example: 
**Upcoming removals**  
**Important features, changes, and deprecations**

**Comment by ektravel** [comment]
@317brian I think we should add this section to the Upgrade notes topic, upgrade-notes.md. I don't think we need to repeat it in the Upgrade notes section of the Release notes topic, release-notes.md.

File: docs/release-info/release-notes.md [hunk 4]
```diff
@@ -53,6 +53,18 @@ If your release note contains images, put the images in the release-info/assets
 For tips about how to write a good release note, see [Release notes](https://github.com/apache/druid/blob/master/CONTRIBUTING.md#release-notes).
 -->
 
+## Upcoming deprecations
```
Comment (techdocsmith) [inline docs/release-info/release-notes.md:56]:
```suggestion
## Upcoming removals
```

**Review by techdocsmith — APPROVED** [review]
LGTM

**Comment by 317brian** [comment]
Copy/pasted the section into upgrade-notes too: 
image

**Review by techdocsmith — APPROVED** [review]

**Pull request merged**